src/catalog/catalog_set.cpp

```diff
@@ -57,50 +62,83 @@ bool CatalogSet::CreateEntry(Transaction &transaction, const string &name, uniqu
 	// now add the dependency set of this object to the dependency manager
 	catalog.dependency_manager->AddObject(transaction, value.get(), dependencies);
 
-	value->child = move(data[name]);
+	value->child = move(entries[entry_index]);
 	value->child->parent = value.get();
 	// push the old entry in the undo buffer for this transaction
 	transaction.PushCatalogEntry(value->child.get());
-	data[name] = move(value);
+	entries[entry_index] = move(value);
+	return true;
+}
+
+bool CatalogSet::GetEntryInternal(Transaction &transaction, idx_t entry_index, CatalogEntry *&catalog_entry) {
+	catalog_entry = entries[entry_index].get();
+	// if it does: we have to retrieve the entry and to check version numbers
+	if (HasConflict(transaction, *catalog_entry)) {
+		// current version has been written to by a currently active
+		// transaction
+		throw TransactionException("Catalog write-write conflict on alter with \"%s\"", catalog_entry->name);
+	}
+	// there is a current version that has been committed by this transaction
+	if (catalog_entry->deleted) {
+		// if the entry was already deleted, it now does not exist anymore
+		// so we return that we could not find it
+		return false;
+	}
 	return true;
 }
 
+bool CatalogSet::GetEntryInternal(Transaction &transaction, const string &name, idx_t &entry_index,
```

Comment:
I'm confused by this: why do we pass the `entry_index` as an arg here, since it looks like it gets overwritten in the method body?

Reply:
Ah, b/c we want to send it back to the client as output-- nm!

Reply:
entry_index in this case is a return value, hence why it is passed as a reference.

Reply:
Whoops, you were faster :)